game-core/src/main/java/games/strategy/engine/framework/startup/launcher/ServerLauncher.java

```diff
@@ -187,44 +175,25 @@ void launchInternal(final Component parent, @Nullable final Void none) {
       });
       stopGame();
     } catch (final RuntimeException e) {
-      final String errorMessage = "Unrecognized error occurred. If this is a repeatable error, "
-          + "please make a copy of this savegame and report to:\n" + UrlConstants.GITHUB_ISSUES;
-      log.log(Level.SEVERE, errorMessage, e);
-      stopGame();
+      // no-op, this is a simple player disconnect, no need to scare the user with some giant stack trace
+      if (!(e.getCause() != null && e.getCause() instanceof ConnectionLostException)) {
+        final String errorMessage = "Unrecognized error occurred. If this is a repeatable error, "
+            + "please make a copy of this savegame and report to:\n" + UrlConstants.GITHUB_ISSUES;
+        log.log(Level.SEVERE, errorMessage, e);
+        stopGame();
+      }
     }
     // having an oddball issue with the zip stream being closed while parsing to load default game. might be
     // caused by closing of stream while unloading map resources.
     Interruptibles.sleep(200);
     // either game ended, or aborted, or a player left or disconnected
-    if (headless) {
-      try {
-        log.info("Game ended, going back to waiting.");
-        // if we do not do this, we can get into an infinite loop of launching a game,
-        // then crashing out, then launching, etc.
-        serverModel.setAllPlayersToNullNodes();
-        final File f1 = AutoSaveFileUtils.getHeadlessAutoSaveFile();
-        if (!f1.exists() || !gameSelectorModel.load(f1)) {
-          gameSelectorModel.resetGameDataToNull();
-        }
-      } catch (final Exception e1) {
-        log.log(Level.SEVERE, "Failed to load game", e1);
-        gameSelectorModel.resetGameDataToNull();
-      }
-    } else {
-      gameSelectorModel.loadDefaultGameNewThread();
-    }
-    if (parent != null) {
-      SwingUtilities.invokeLater(() -> JOptionPane.getFrameForComponent(parent).setVisible(true));
-    }
+    launchAction.handleGameInterruption(gameSelectorModel, serverModel);
     serverModel.setServerLauncher(null);
     serverModel.newGame();
     if (inGameLobbyWatcher != null) {
       inGameLobbyWatcher.setGameStatus(GameDescription.GameStatus.WAITING_FOR_PLAYERS, null);
     }
-    if (headless) {
-      // tell headless server to wait for new connections:
-      HeadlessGameServer.waitForUsersHeadlessInstance();
-    }
+    launchAction.onGameInterrupt();
```

Comment:
`handleGameInterruption` and `onGameInterrupt` seem like they are basically the same method, they are always called together, can we combine the two? It does seem there are sequencing issues, perhaps those can be solved?